File: problem2/fixed/B01138859.py
def maxValidWindowSum(nums, k):
    first = True
    sum = 0
    ans = 0
    numNeg = 0
    left = 0
    right = 0
    while right < len(nums):
        if right - left < k:
            if nums[right] < 0:
                numNeg += 1
                sum += nums[right]
                right += 1
            else:
                if numNeg <= 1:
                    if first:
                        ans = sum
                    else:
                        ans = max(ans,sum)
                        first = False
                        if nums[left] < 0:
                            numNeg -= 1
                            if nums[right] < 0:
                                numNeg += 1
                                sum -= nums[left]
                                sum += nums[right]
                                left += 1
                                right += 1
                                if numNeg <= 1:
                                    if first:
                                        ans = sum
                                    else:
                                        ans = max(ans,sum)
                                        return ans

File: problem1/java/B01138859.java
public class Solution {
public static int maxValidWindowSum(int[] nums, int k)
    {
        boolean first = true;
        int sum = 0;
        int ans = 0;
        int numNeg = 0;
        int left = 0;
        int right = 0;
        while (right < nums.length)
        {
            if (right - left < k)
            {
                if (nums[right] < 0)
                {
                    numNeg++;
                }
                sum += nums[right];
                right++;
            }
            else
            {
                if (numNeg <= 1)
                {
                    if (first)
                    {
                        ans = sum;
                    }
                    else
                    {
                        ans = Math.max(ans,sum);
                    }
                    first = false;
                }
                if (nums[left] < 0)
                {
                    numNeg--;
                }
                if (nums[right] < 0)
                {
                    numNeg++;
                }
                sum -= nums[left];
                sum += nums[right];
                left++;
                right++;
            }
        }
        if (numNeg <= 1)
        {
            ans = Math.max(ans,sum);
        }
        return ans;
    }
}